documentation/src/docs/controller-service.md

```diff
@@ -154,296 +151,276 @@ Controller Process Diagram
- the stream cut in truncation record.
+ This corresponds to the `StreamCut` which was last used to truncate the
+ given Stream. All Stream Segment queries superimpose the truncation
+ record and return Stream Segments that are strictly greater than or equal to
+ the `StreamCut` in truncation record.
+
+#### Sealed Segments Maps
+Once the Stream Segments are sealed, Controller needs to store additional information about the Stream Segment. Presently, we have two types of information:
 
-#### Sealed Segments Record
- Since the segment table is append only, any additional information
- that we need to persist when a segment is sealed is stored in sealed
- segments record. Presently, it simple contains a map of segment number
- to its sealed size.
+ - Epoch, the Stream Segment was sealed in.
+ - Size of the Stream Segment at the time of sealing.
+
+ These records have two different characteristics and are used in different types of queries. For example;
+
+ 1. Sealing epoch is important for querying successor Stream Segments. For each Stream Segment we store its sealing epoch directly in the metadata store.
+ 2. Stream Segment sizes are used during truncation workflows. For sealed sizes, we store it in a map of Segment to size at the time of sealing.
+ 3. Successor queries are performed on a single Stream Segment whereas truncation workflows work on a group of Stream Segments.
+
+ This ensures that during truncation we are able to retrieve sealed sizes for multiple Stream Segments with minimal number of calls into underlying metadata store. Since we could have arbitrarily large number of Stream Segments that have been sealed away, we cannot store all of the information in a single map and hence we shard the map and store it. The sharding function we use is to hash the creation epoch and get the shard number.
 
 The following are the Transaction Related metadata records:
 
-   - **Active Transactions**: Each new Transaction is created under this znode. This stores metadata
- corresponding to each Transaction as *ActiveTransactionRecord*. Once a
+   - **Active Transactions**: Each new Transaction is created under the znode. This stores metadata
+ corresponding to each Transaction as *Active Transaction Record*. Once a
  Transaction is completed, a new node is created under the global
- Completed Transaction znode and removed from under the stream specific
- _active_ Transaction node.
+ _Completed Transaction_ znode and removed from under the Stream specific
+ _Active Transaction_ node.
 
-   - **Completed Transactions**: All completed transactions for all streams are moved under this single
+   - **Completed Transactions**: All completed transactions for all Streams are moved under this single
  znode upon completion (via either commit or abort paths). We can
  subsequently garbage collect these values periodically following any
  collection scheme we deem fit. We have not implemented any scheme at
```

Comment:
We have implemented this scheme. The text needs to be updated.

Reply:
@shiveshr Any updates on this As well `completed transactions`

Reply:
```

Completed Transactions: All completed transactions for all Streams are moved under a separate znode upon completion (via either commit or abort paths). The completion status of transaction is recorded under this record. To avoid proliferation of stale transaction records, we provide a cluster level configuration to specify the duration for which a completed transaction's record should be preserved. Controller periodically garbage collects all transactions that were completed before the aforesaid configured duration.
```

Reply:
Accepted Suggestion